qiskit_experiments/data_processing/processor_library.py

```
@@ -14,29 +14,62 @@
-    normalize: bool = True,
-) -> DataProcessor:
+def get_processor(experiment_data: ExperimentData, index: int = -1) -> DataProcessor:
     """Get a DataProcessor that produces a continuous signal given the options.
 
     Args:
-        meas_level: The measurement level of the data to process.
-        meas_return: The measurement return (single or avg) of the data to process.
-        normalize: Add a data normalization node to the Kerneled data processor.
+        experiment_data: The experiment data that holds all the data and metadata needed
+             to determine the data processor to use to process the data for analysis.
+        index: The index of the job for which to get a data processor.
 
     Returns:
-        An instance of DataProcessor capable of dealing with the given options.
+        An instance of DataProcessor capable of processing the data for the corresponding job.
 
     Raises:
         DataProcessorError: if the measurement level is not supported.
     """
+
+    run_options = experiment_data.metadata["job_metadata"][index].get("run_options", {})
+    analysis_options = experiment_data.metadata["job_metadata"][index].get("analysis_options", {})
+
+    physical_qubits = experiment_data.metadata["physical_qubits"]
+    num_qubits = len(physical_qubits)
+    t1_values = [
+        experiment_data.backend.properties().qubit_property(physical_qubit)["T1"][0]
+        for physical_qubit in physical_qubits
+    ]
+
+    meas_level = run_options.get("meas_level", MeasLevel.CLASSIFIED)
+    meas_return = run_options.get("meas_return", "avg")
+    normalize = analysis_options.get("normalization", True)
+
+    init_qubits = run_options.get("init_qubits", True)
+    memory = run_options.get("memory", False)
+    rep_delay = run_options.get("rep_delay", None)
+
+    # restless data processing.
+    restless = False
+    if rep_delay:
+        if [rep_delay / t1_value < 0.1 for t1_value in t1_values] == [True] * num_qubits:
```

Can we make this 0.1 factor an analysis options with a default value. E.g.
```python
restless_threshold = analysis_options.get("restless_threshold", 1)
```
also I think that 0.1 is a bit low. Perhaps 1 is still okay.

---

Also, if we have determined that the data is restless then we should raise an error if there is no memory. E.g.
```python
if restless and not memory:
    raise DataProcessingError(f"Run options suggest restless data but no single-shot memory is present.")
```
We also need to make sure that the memory contains single-shots and not averaged data.

---

See 4280a96.